### PR TITLE
Add utility class

### PR DESCRIPTION
**Pull request by Dunn-Austen**
**What's this PR do?*
-This PR establishes a new class, Utility, with a corresponding file within the SRC folder
-Populates the Utility class with refactored methods that correspond to redundant methods across multiple classes. You should know that the function names have been changed to be more specific and the code within the refactored functions have been change
-The function for findUserData() has been built into our Utility class property, eliminating the need for the function itself
-Imports Utility into the scripts file

**Where should the reviewer start?**
- Check out the new Utility file

**How should this be manually tested?**
-This is a next step. The testing suite will need to be built out

**What are the relevant tickets?**
-This PR will resolve issue #24 

**Questions:**
- We'll need to eliminate the original methods from their respective class files.
- We'll need to purge and polish the original testing suites for the respective classes after pruning their class methods
- We'll need to build out the testing suite for Utility
- We'll need to investigate the original function invocations and alter them to match the new names we've drafted for the refactored functions

### CODE REVIEW

File: src/Utility.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Utility {
+  constructor(dataSet, userID) {
+    this.dataSet = dataSet;
+    this.userID = userID;
+    this.singleUserData = this.dataSet.filter(stat => stat.userID === this.userID);
+  }
+
+  returnWeekDatesOnly() {
+    return [...this.singleUserData].splice(-7 * week, 7).map(day => day.date);
+  }
```
Comment (hfaerber) [inline src/Utility.js]: I think we will need to include a 'week' parameter here since the method is referencing it.

File: src/scripts.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 
 import $ from 'jquery';
 
+import Utlity from "./Utility";
```
Comment (hfaerber) [inline src/scripts.js]: Just a tiny typo here.  Utlity > Utility

**Comment by hfaerber** [comment]
@Dunn-Austen  - I noted two little things that we'll want to correct before we merge but this looks great.  Thank you for leading us through building this out!  I really appreciate that we all had the opportunity to make sure we had our heads wrapped around it and ensure that the naming conventions make sense to all of us.   Really great job.

**Comment by Dunn-Austen** [comment]
Hey Heather, 

Appreciate the feedback. I'm going to include these changes in my next PR. Good eye & please go ahead and merge this request

**Comment by hfaerber** [comment]
> Hey Heather,
> 
> Appreciate the feedback. I'm going to include these changes in my next PR. Good eye & please go ahead and merge this request

Sounds good!  I'll merge then we can each correct locally to proceed since they're both such tiny changes.   @NaomiKC92  - when you pull down, you'll want to adjust these two tiny things before working on updating the Sleep class methods/testing.

**Pull request merged**